src/services/aws_macros.ts

```diff
@@ -59,9 +67,26 @@ export class AWS {
   secretsClient: SecretsManager;
   cloudfrontClient: CloudFront;
   memoryDBClient: MemoryDB;
+  slowRetryStrategy: StandardRetryStrategy;
 
   constructor(config: AWSConfig) {
+    // declare an specific slow retry strategy, to reuse in slow apis
+    this.slowRetryStrategy = new StandardRetryStrategy(async () => SLOW_STRATEGY_RETRIES, {
+      delayDecider: (_, attempts) =>
+        Math.floor(
+          Math.min(
+            SLOW_STRATEGY_MAXIMUM_RETRY_DELAY,
+            Math.random() * 2 ** attempts * SLOW_STRATEGY_BASE_DELAY,
```

Comment:
Slapping in the `Math.random()` in this way causes a highly variable retry delay from 0ms to `(2^attempts) * SLOW_STRATEGY_BASE_DELAY`ms. Perhaps this would be a bit better?

```suggestion
            (Math.random() + 2) ** attempts * SLOW_STRATEGY_BASE_DELAY,
```

Then it will always take at least `(2^attempts) * SLOW_STRATEGY_BASE_DELAY` in time, but could take up to `(3^attempts) * SLOW_STRATEGY_BASE_DELAY)`

Reply:
i just picked the strategy used by aws, but increasing the delay time. We can optimize that for the future, but if we modify we could create a default strategy for all clients.